packages/dashboard/src/customization/api.ts

```diff
@@ -66,8 +66,14 @@ const resetMaps = () => {
   clearArr(ComponentLibraryComponentOrdering);
 };
 
-export const setupDashboardPlugins = (plugins: DashboardPlugin[]) => {
-  resetMaps();
+export const useDashboardPlugins = (plugins: DashboardPlugin[]) => {
+  const [isLoaded, setLoaded] = useState(false);
 
-  plugins.forEach((plugin) => plugin.install({ registerWidget }));
+  useEffect(() => {
```

Comment:
I think the useeffect is actually causing more problems since it runs after render. Since we use this hook in the root component, this means that we won't have any plugins installed when the components that need them render. For instance Grid and Component Palette. We can just get rid of the useeffect. Otherwise, we'll have to delay rendering until the second pass every time. Also we need to make use of the feature flags in this hook.

Reply:
done